lib/wallet/integration/src/Test/Integration/Scenario/API/Shared/Transactions.hs

```diff
@@ -2587,22 +2594,282 @@ spec = describe "SHARED_TRANSACTIONS" $ do
                 , expectField (#balance . #available)
                     (.> (Quantity faucetUtxoAmt))
                 ]
+
+    it "SHARED_TRANSACTIONS_LIST_05 - filter address output side" $
+        \ctx -> runResourceT $ do
+        let minUTxOValue' = minUTxOValue (_mainEra ctx)
+        let a1 = minUTxOValue'
+        let a2 = 2 * minUTxOValue'
+        let a3 = 3 * minUTxOValue'
+        let a4 = 4 * minUTxOValue'
+        let a5 = 5 * minUTxOValue'
+        (wSrc, ApiSharedWallet (Right wDest)) <-
+             (,) <$> fixtureSharedWalletWith ctx 100_000_000 <*> emptySharedWallet ctx
+
+        -- initially empty wallet wDest should have two txs on address 0,
+        -- one tx on address 1 and three txs on address 2
+        sendAmtToAddr ctx wSrc wDest a1 0
+        sendAmtToAddr ctx wSrc wDest a2 0
+        sendAmtToAddr ctx wSrc wDest a3 1
+        sendAmtToAddr ctx wSrc wDest a1 2
+        sendAmtToAddr ctx wSrc wDest a4 2
+        sendAmtToAddr ctx wSrc wDest a5 2
+
```

Comment:
```suggestion
let listTransactionsFilteredByAddress wallet =
        Link.listTransactions' @'Shared wallet
            Nothing Nothing Nothing Nothing Nothing
```
In order to make the integration tests easier to read, I favor factoring out this common usage pattern.

Reply:
good suggestion! Refactored this for all styles